packages/gatsby-source-contentful/src/cache-image.js

```diff
@@ -45,24 +47,36 @@ module.exports = async function cacheImage(store, image, options) {
   const { name, ext } = parse(fileName)
   const absolutePath = resolve(CACHE_DIR, `${name}-${optionsHash}${ext}`)
 
+  // If there's already a request in flight await it, then return the path
+  const inFlight = inFlightImageCache.get(absolutePath);
+  if (inFlight) {
+    await inFlight;
+    return absolutePath;
+  }
+  
+  // If the file does not exist download it, put the promise in the cache, and await
   const alreadyExists = await pathExists(absolutePath)
-
   if (!alreadyExists) {
-    const previewUrl = `http:${url}?${params.join(`&`)}`
-
-    const response = await axios({
-      method: `get`,
-      url: previewUrl,
-      responseType: `stream`,
-    })
+    const downloadPromise = await new Promise((resolve, reject) => {
```

Comment:
There is `await` used in promise executor so we need to add `async` to function 
```suggestion
    const downloadPromise = await new Promise(async (resolve, reject) => {
```